cmake/BuildLibCURL.cmake

```diff
@@ -205,8 +205,9 @@ macro(curl_build)
         CONFIGURE_COMMAND
```

Comment:
There is an impression that `curl` and `c-ares` are still built using nested `make` instead of `ninja` even if there is outer `ninja` initiated whole build process. We need to make sure we properly pass "generator" (`-G ninja`) down to libcurl.

But, taking into account, that we already have good performance boost - this might be resolved elsewhere, later.

Reply:
Right for libcurl - `ExternalProject_Add` used custom `CONFIGURE_COMMAND` and cmake there was not aware about using `ninja` instead of default `make`. Fixed with diff below:
```diff
--- a/cmake/BuildLibCURL.cmake
+++ b/cmake/BuildLibCURL.cmake
@@ -207,6 +207,7 @@ macro(curl_build)
                 ${LIBCURL_CMAKE_FLAGS}
         BUILD_COMMAND cd <BINARY_DIR> && make
         INSTALL_COMMAND cd <BINARY_DIR> && make install
+        CMAKE_GENERATOR ${CMAKE_GENERATOR}
         BUILD_BYPRODUCTS ${LIBCURL_INSTALL_DIR}/lib/libcurl.a)
 
     add_library(bundled-libcurl STATIC IMPORTED GLOBAL)
```

`CMAKE_GENERATOR` override the CMake generator used for the configure step. Without this option, the same generator as the main build will be used. This option is ignored if a custom configure command has been specified with the CONFIGURE_COMMAND option. See https://cmake.org/cmake/help/latest/module/ExternalProject.html

ExternalProject_Add for `c-ares` doesn't customize `BUILD_COMMAND`.